lib/charms/layer/apache_bigtop_base.py

```diff
@@ -39,12 +40,32 @@ def site_yaml(self):
     def __init__(self):
         self.bigtop_dir = '/home/ubuntu/bigtop.release'
         self.options = layer.options('apache-bigtop-base')
-        self.bigtop_version = self.options.get('bigtop_version')
-        self.bigtop_apt = self.options.get(
-            'bigtop_repo-{}'.format(utils.cpu_arch()))
-        self._bigtop_base = Path(self.bigtop_dir) / self.bigtop_version
-        self._site_yaml = self.bigtop_base / self.options.get(
-            'bigtop_hiera_siteyaml')
+        self.bigtop_version = self.options['bigtop_version']
+        self._bigtop_base = Path(self.bigtop_dir) / 'bigtop-{}'.format(
+            self.bigtop_version)
+        self._site_yaml = Path(self.bigtop_base) / self.options[
+            'bigtop_hiera_siteyaml']
+
+        dist_name, _, dist_series = platform.linux_distribution()
+        # NB: xenial repo is not currently available, and ppc64le only works
+        # with vivid. Force vivid on ppc and trusty on everyone else for now:
+        #   http://paste.ubuntu.com/18185418/
+        repo_arch = utils.cpu_arch()
+        if repo_arch == "ppc64le":
+            dist_series = "vivid"
+            # The 'le' and 'el' are swapped due to historical debian awfulness.
+            #   https://lists.debian.org/debian-powerpc/2014/08/msg00042.html
+            repo_arch = "ppc64el"
+        else:
+            dist_series = "trusty"
```

Comment:
Yes, even though our OS is now xenial-only, the repo that we use to install bigtop-1.1.0 only offers "trusty" for x86 and "vivid" for ppc64le.  I assume this will change once 1.2.0 is released and bigtop offers official xenial repos.